src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.NumericValueSet.cs

```diff
@@ -135,13 +135,13 @@ static bool allInterval(Interval interval, BinaryOperatorKind relation, T value,
                                     return false;
                                 case GreaterThan when tc.Related(GreaterThan, minValue, value):
                                     return true;
-                                case LessThanOrEqual when tc.Related(LessThanOrEqual, value, minValue):
+                                case LessThanOrEqual when tc.Related(LessThan, value, minValue):
```

Comment:
>LessThan [](start = 69, length = 8)

Is `OrEqual` not needed?

Reply:
It would not be correct.  Consider a mixed interval [1..10] and a value of 1.  We don't know if all of the values in the interval are LessThanOrEqual to value.  If the interval contains only the number 1, then they are all LessThanOrEqual to the value and we should return true.  Otherwise we should return false.  So `tc.Related(LessThan, value, minValue)` would not be the correct test here to know if we should return false.

Which only goes to prove that this method and the corresponding `anyInterval` above are extremely subtle.  I added a bunch of tests to expose any off-by-one errors in these conditions.

---
In reply to: [392361775](https://github.com/dotnet/roslyn/pull/42313#discussion_r392361775) [](ancestors = 392361775)